File: fathersDay/Dsa/Day 24 (129. Sum Root to Leaf Numbers)/Solution.py
class Solution:
    class TreeNode:
        def __init__(self, val=0, left=None, right=None):
            self.val = val
            self.left = left
            self.right = right
    def sumNumbers(self, root: TreeNode) -> int:
        def dfs(node, current_sum):
            if not node:
                return 0
            current_sum = current_sum * 10 + node.val
            if not node.left and not node.right:
                return current_sum
            return dfs(node.left, current_sum) + dfs(node.right, current_sum)
        return dfs(root, 0)
solution = Solution()
# Example usage:
root = solution.TreeNode(1)
root.left = solution.TreeNode(2)
root.right = solution.TreeNode(3)
root.left.left = solution.TreeNode(4)
root.left.right = solution.TreeNode(5)
result = solution.sumNumbers(root)
print(result)  # Output: 262, as the root-to-leaf numbers are 124 and 125, and their sum is 124 + 125 = 249

File: fathersDay/Dsa/Day 24 (129. Sum Root to Leaf Numbers)/Solution.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int val) { this.val = val; }
}

class Solution {
    public int sumNumbers(TreeNode root) {
        return dfs(root, 0);
    }

    private int dfs(TreeNode node, int currentSum) {
        if (node == null) {
            return 0;
        }
        currentSum = currentSum * 10 + node.val;
        if (node.left == null && node.right == null) {
            return currentSum;
        }
        return dfs(node.left, currentSum) + dfs(node.right, currentSum);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        int result = solution.sumNumbers(root);
        System.out.println(result); // Output: 25
    }
}
